src/mlpack/methods/kde/kde_rules_impl.hpp

```diff
@@ -150,6 +173,13 @@ Score(const size_t queryIndex, TreeType& referenceNode)
     // Don't explore this tree branch.
     score = DBL_MAX;
 
+    // Subtract used error tolerance or add extra available tolerace from this
+    // prune.
+    if (alreadyDidRefPoint0)
+      accumError(queryIndex) -= (refNumDesc - 1) * (bound - 2 * errorTolerance);
+    else
+      accumError(queryIndex) -= refNumDesc * (bound - 2 * errorTolerance);
```

Comment:
The same ternary operator trick could be used here, but again, up to you how you want to handle it.